source/plugin/data/serialization.rst

```diff
@@ -168,22 +168,14 @@ For this example we will use :javadoc:`DataFormats#JSON` to translate a ItemStac
 
 .. code-block:: java
 
-    public Optional<ItemStack> fromJsonFormat(InputStream inputStream){
-        DataFormat df = DataFormats.JSON;
-        try {
-            Optional<DataContainer> dataContainerOptional = Optional.of(df.readFrom(inputStream));
-            if (dataContainerOptional.isPresent()){
-                DataContainer dataContainer = dataContainerOptional.get();
-                final Optional<DataBuilder<ItemStack>> itemStackDataBuilder = Sponge.getDataManager().getBuilder(ItemStack.class);
-                if (itemStackDataBuilder.isPresent()){
-                    DataBuilder<ItemStack> itemStackBuilder =  itemStackDataBuilder.get();
-                    return itemStackBuilder.build(dataContainer);
-                    
-                }
+    public static Optional<ItemStackSnapshot> getItemStackSnapshotFromDataContainer(DataContainer dataContainer){
+        final Optional<DataBuilder<ItemStackSnapshot>> dataBuilderOptional =            Sponge.getDataManager().getBuilder(ItemStackSnapshot.class);
```

Comment:
All of the data manager stuff is unnecessary, all you need to do is call `dataContainer.getSerializable(ItemStackSnapshot.class, query)` or however the parameters are ordered.